clients/privacy-center/cypress/e2e/should-show-experience.cy.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  CONSENT_COOKIE_NAME,
+  NoticeConsent,
+  PrivacyExperience,
+} from "fides-js";
+
+import { mockCookie } from "../support/mocks";
+import { OVERRIDE, stubConfig, stubTCFExperience } from "../support/stubs";
+
+describe("Fides.shouldShowExperience()", () => {
+  interface TestCaseOptions {
+    fixture:
+      | "experience_modal.json"
+      | "experience_banner_modal.json"
+      | "experience_tcf_minimal.json"
+      | "experience_none.json";
+    savedConsent: boolean;
+    expiredTcfVersionHash?: boolean;
+    shouldShowExperience: boolean;
+  }
```

Comment:
This test fixtures modify three inputs:

1. `fixture`: the type of experience to test
2. `savedConsent`: whether or not the user has a previously saved consent cookie
3. `expiredTcfVersionHash`: special case for TCF, whether the user should have a previously saved consent cookie but with an expired version hash (that doesn't match the latest experience)

It then asserts to see if `Fides.shouldShowExperience()` returns the expected result of `shouldShowExperience` in the param.